### PR TITLE
feat: numaflow package

### PR DESCRIPTION
**Pull request by veds-g**
- Added components to render separate views to be consumed by the package
- Changed UI router to use query-based params for routing
- Enabled typescript checks for all files
- Fixed typescript errors in existing components
- Updated `package.json` to setup for package creation

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -1,7 +1,26 @@
 {
-  "name": "ui",
-  "version": "0.1.0",
-  "private": true,
+  "name": "numaflow-ui-test",
```
Comment (vigith) [inline ui/package.json:2]: test?

File: ui/package.json [hunk 2]
```diff
@@ -1,7 +1,26 @@
...
+  "version": "0.2.2",
+  "description": "Numaflow UI",
+  "main": "./dist/cjs/src/index.js",
+  "module": "./dist/esm/src/index.js",
+  "types": "./dist/esm/src/index.d.ts",
+  "license": "Apache-2.0",
+  "author": "Vedant Gupta",
```
Comment (vigith) [inline ui/package.json:9]:
```suggestion
  "author": "Numaflow Developers",
```

**Review by vigith — COMMENTED** [review]
@mshakira could you please review?

File: ui/package.json (hunk 1 above)
Comment (veds-g) [reply ui/package.json:2]: currently, we have not created an account for numaflow on npm, so I am using my personal account, thus the test name and author, once we have the account I will update this.

Comment (vigith) [reply ui/package.json:2]: can you please open an issue to track this?

Comment (veds-g) [reply ui/package.json:2]: created - https://github.com/numaproj/numaflow/issues/1585

File: ui/src/components/pages/Pipeline/partials/Graph/partials/EdgeInfo/index.tsx [hunk 3]
```diff
@@ -81,16 +81,18 @@ export default function EdgeInfo(props: EdgeInfoProps) {
...
-                    return (
-                      <TableRow key={`edge-watermark-${idx}`}>
-                        <TableCell>{idx}</TableCell>
-                        <TableCell>
-                          {wmVal} ({new Date(wmVal).toISOString()})
-                        </TableCell>
-                      </TableRow>
-                    );
-                  })}
+                  edge.data.edgeWatermark.watermarks.map(
```
Comment (mshakira) [inline ui/src/components/pages/Pipeline/partials/Graph/partials/EdgeInfo/index.tsx:84]: please check for null

**Review by mshakira — APPROVED** [review]

**Pull request merged**